content/docs/grammar-language.md

```diff
@@ -0,0 +1,371 @@
+    ```
+    paragraph "The expression group " 3 "was interrupted"
+    ```
+    will throw an error since the `STRING` expressions are not continuous. It is however possible to interrupt and resume a sequence of expressions by using [hidden terminal symbols](#hidden-terminal-symbols):
+    ```
+    paragraph "expression one" /* comment */ "expression two"
+    ```
+    The above example will be successfully parsed.
+
+3. `?=` is used to assign a **value to a property of type boolean**. The value of the property of type `boolean` is set to `true` if the right part of the assignment is consumed by the parser.
+    ```
+    Employee:
+        'employee' name=ID (remote?='remote')?
+    ```
+    Here the value of the property `remote` will be set to true if the keyword `remote` appears in the document.
+
+#### Cross-reference
+With Langium, you can declare *cross-references* directly in the grammar. *Cross-reference* allows to reference an object of a given type. The syntax is:
+```
+property=[Type:TOKEN]
+```
+The `property` will be a reference to an object of type `Type` identified by the token `TOKEN`. If the `TOKEN` is omitted, the parser assumes that it is matching the terminal rule `ID`.
+```
+Person:
+    'person' name=ID;
+Greeting:
+    'Hello' person=[Person:ID] '!';
+```
+The `Person` in square brackets does not refer to the parser rule `Person` but instead refers to an object of type `Person`. It will successfully parse a document like:
+```
+person Bob
+Hello Bob !
+```
+but will fail to parse:
+```
+person Bob
+Hello Sara !
+```
+because a `Person` object with the ID of 'Sara' has not been instantiated even though 'Sara' is a valid `ID`.
+
+#### Unordered Groups
+[NOTE] Unordered group are currently not supported but seems like it's being worked on. I think this is still valuable info that could be added to the documentation at a later stage if needed.
+
+By default, a parser rule has to be implemented in the exact order it is declared.
+```
+Person:
+    'person' name=ID age=INT
+```
+Here a `Person` object **needs** to first declare the property `name` then `age`.
+```
+person Bob 25
+```
+will successfully be parsed to an object of type `Person` while
+```
+person 25 Bob
+```
+will throw an error.
+
+However, it is possible to declare a group of properties in an unordered fashion using the `&` operator
+```
+Person:
+    'person' name=ID & age=INT
+```
+will now allow `name` and `age` to be declared in any order.
+```
+person 25 Bob
+```
+will then successfully create an object of type `Person`.
+
+Cardinality (?,*,+ operators) also applies to unordered group. Please note that assignments with a cardinality of `+` or `*` have to appear continuously and cannot be interrupted by an other assignment and resumed later.
```

Comment:
Minor: Escape usage of `*` using `\*`. This breaks markdown formatting.

Reply:
It looks fine on the Netlify preview